docs/CONTRIBUTING.md

```diff
@@ -115,6 +115,24 @@ Use `nox` to run `pytest` to run tests:
 nox
 ```
 
+### Running tests without slow integration tests
+
+Some integration tests (like the full visualization workflow test) are marked as
+`slow`. To run tests without these slow tests:
+
+```bash
+pytest -m "not slow"
+```
+
+Or with nox:
+
+```bash
+nox -s tests -- -m "not slow"
+```
+
+This is useful for faster feedback during local development. The full test suite
+(including slow tests) runs in CI.
```

Comment:
This isn't quite right - in CI we only execute the unit tests - those in the `tests/unit` dir. This is because the integration tests require local k8s clusters with the stack installed. 

I would love to get some kind of automated integration testing setup as part of our CI though - it might be worth thinking more about.